IO.Eventuate.Tram/JsonMapper.cs

```diff
@@ -6,39 +6,36 @@
  */
 
 using System;
-using Newtonsoft.Json;
-using Newtonsoft.Json.Serialization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
 
 namespace IO.Eventuate.Tram
 {
-	public static class JsonMapper
-	{
-		public static readonly JsonSerializerSettings JsonSerializerSettings = new JsonSerializerSettings
-		{
-			// Don't use CamelCasePropertyNamesContractResolver
-			// The CamelCasePropertyNamesContractResolver uses an internal cache that is shared between instances,
-			// so sometimes dictionary keys were getting camel cased if the wrong settings got cached.
-			ContractResolver = new DefaultContractResolver
-			{
-				NamingStrategy = new CamelCaseNamingStrategy()
-			},
-			MissingMemberHandling = MissingMemberHandling.Ignore,
-			NullValueHandling = NullValueHandling.Ignore
-		};
-		
-		public static string ToJson(object o)
-		{
-			return JsonConvert.SerializeObject(o, JsonSerializerSettings);
-		}
+    public static class JsonMapper
+    {
+        public static readonly JsonSerializerOptions JsonSerializerOptions = new JsonSerializerOptions
+        {
+            // Use camel case for property names
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            // Ignore null values during serialization
+            DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+            // Make property name matching case-insensitive during deserialization
+            PropertyNameCaseInsensitive = true
```

Comment:
I'm unsure as to whether this should be case sensitive or insensitive. The original intent of the .NET port was to match the behavior of the Java implementation, so perhaps we should try to match what it does by default.

@cer: Do you know if the Java eventuate-tram is case-sensitive or insensitive for property names during deserialization?

Reply:
It's case sensitive although it is configurable: https://fasterxml.github.io/jackson-databind/javadoc/2.5/com/fasterxml/jackson/databind/MapperFeature.html#ACCEPT_CASE_INSENSITIVE_PROPERTIES